### PR TITLE
Use sbt-ci-release

### LINKED ISSUE #341: Consider updating packaging and coordinates?
As discussed [here](https://github.com/typelevel/general/issues/101), this repository is now in the Typelevel organization. It's not strictly necessary to move the project from `io.catbird` to `org.typelevel`, but I think it'd be a good idea. This would involve two changes:

* Change the publishing coordinates from `io.catbird` to `org.typelevel`. This is easy, simplifies publishing rights, and imposes a very small burden on users (just updating the dependency in their build config).
* Change the packaging from `io.catbird.*` to `org.typelevel.catbird.*`. This is more of a burden on users, but is a completely mechanical change.

We could technically just do the first but not the second, but I also think it'd be good to stay consistent, and I'd like to get rid of the io domain at some point (for [various reasons](https://gigaom.com/2014/06/30/the-dark-side-of-io-how-the-u-k-is-making-web-domain-profits-from-a-shady-cold-war-land-deal/)).

### PR DESCRIPTION
**Pull request by bpholt**
Fixes #341 and #380.

We still need someone to [generate a gpg keypair](https://github.com/sbt/sbt-ci-release#gpg) and [update the GitHub Secrets for the repository](https://github.com/sbt/sbt-ci-release#secrets). (I'm happy to do this, but I would need to be made an admin to do so.)

### CODE REVIEW

File: util/src/main/scala/org/typelevel/catbird/util/asyncstream.scala
Comment (bpholt) [inline util/src/main/scala/org/typelevel/catbird/util/asyncstream.scala:24]: I made this change because with the additional compiler warnings, the compiler was complaining that the `implicit A: Eq[A]` was unused. This is a 2.12/2.13 cross-compatible way to check equality using the `Eq[A]` instance.

File: effect/src/test/scala/org/typelevel/catbird/util/effect/RerunnableContextShiftSuite.scala
Comment (bpholt) [inline effect/src/test/scala/org/typelevel/catbird/util/effect/RerunnableContextShiftSuite.scala:45]: `poolName3` was previously unused. I'm not 100% sure if this is the right test, but it passes.

File: effect/src/main/scala/org/typelevel/catbird/util/effect/package.scala
Comment (bpholt) [inline effect/src/main/scala/org/typelevel/catbird/util/effect/package.scala:20]: Explicitly return unit to indicate that we're ok discarding the `Future` returned by `fa.respond`

File: effect3/src/main/scala/org/typelevel/catbird/util/effect/FutureInstances.scala
Comment (bpholt) [inline effect3/src/main/scala/org/typelevel/catbird/util/effect/FutureInstances.scala:18]: `Any` was the inferred type, but with sbt-tpolecat, an inferred `Any` ends up as an error because it is often incorrect.

**Comment by bpholt** [comment]
I think we should probably also add a [Scala Steward artifact migration](https://github.com/scala-steward-org/scala-steward/blob/master/modules/core/src/main/resources/artifact-migrations.conf) and Scalafix rewrite rule to reduce the impact on users.

**Review by felixbr — APPROVED** [review]

**Comment by felixbr** [comment]
Just one small thing: The CI pipeline currently only runs for 2.12.x and 2.13.x. If it's easy to do, it would be nice to run the `effect3` module with `3.0` so regressions there are also caught in PRs.

**Comment by bpholt** [comment]
In light of [Travis moving away from Scala OSS](https://meta.plasm.us/posts/2021/11/17/scala-open-source/), it seems we need to identify someone to take over the publishing and GPG key management for this project. @felixbr are you the right person for that? My thought is to identify at least one person and then work with Travis for them to take over.

**Comment by bpholt** [comment]
@felixbr friendly bump on [my question above](https://github.com/typelevel/catbird/pull/384#issuecomment-971779138) 🙂

**Comment by felixbr** [comment]
Much like Travis I'm extremely disappointed in the Scala leadership (Odersky but also others) and last year I decided to no longer do any open-source work for Scala unless it's directly needed by my company.

Life is too short to provide free labor for terrible people.

Personally I'd be happy if you would take over this project as a maintainer since you already made some good improvements and it's bad that your contributions are blocked by my decision.
The project is under typelevel now, so a semi-automated release process will hopefully also help to distribute work between more shoulders.

**Comment by armanbilge** [comment]
@bpholt I'd be happy to help move the build onto [sbt-typelevel](https://github.com/typelevel/sbt-typelevel/).

> We still need someone to generate a gpg keypair and update the GitHub Secrets for the repository. (I'm happy to do this, but I would need to be made an admin to do so.)

We're in the process of installing org-wide secrets for these.

**Comment by bpholt** [comment]
We're planning to migrate this repo to sbt-typelevel, but it seems cleaner to do so in a separate PR. I'm going to merge this but we'll hold off on releasing anything until that switch has been made.

**Pull request merged**
